includes/class-wc-post-types.php

```diff
@@ -312,6 +312,7 @@ public static function register_post_types() {
 					'has_archive'         => $has_archive,
 					'show_in_nav_menus'   => true,
 					'show_in_rest'        => true,
+					'menu_icon'           => 'data:image/svg+xml;base64,' . base64_encode( str_replace( '<svg', '<svg fill="black"', get_gridicon( 'gridicons-product' ) ) ), // SVG 'fill' attribute must be set for menu colors to apply correctly.
```

Comment:
The `svg` tag here needs to have a defined `fill` attribute to get the admin menu color scheme to apply correctly. I could edit the svg directly, but the changes would get overwritten if the Gridicons library got updated. The `str_replace` isn't elegant, but it does the job.